spec/unit/provider/registry_key_spec.rb

```diff
@@ -111,6 +111,13 @@
         @provider.load_current_resource
         @provider.action_create
       end
+      it "should not set the value if the key exists but the datatype of data does not match" do
```

Comment:
@btm, 

This test case is applicable when the type is  `dword, qword or dword_big_endian`  and the datatype of the data differs(the data value set is in integer and the data value passed by user is string)   
```
current_value = {:name => 'Enabled', :type => :dword, :data => 4294967295}   #current value set in registry key
value = {:name => 'Enabled', :type => :dword, :data => "4294967295"}         #value passed by user which needs to be set
```
So, updating the description as follows:- 

```
it "does not make a change when the key exists and the type matches, but the datatype of data value differs"
```